CMakeLists.txt

```diff
@@ -30,6 +30,13 @@ option(USE_NCCL "Use NCCL library" ON)
 option(USE_SENTENCEPIECE "Download and compile SentencePiece" ON)
 option(USE_STATIC_LIBS "Link statically against non-system libs" OFF)
 option(GENERATE_MARIAN_INSTALL_TARGETS "Generate Marian install targets (requires CMake 3.12+)" OFF)
+option(M32_BINARIES "Generate 32bit binaries even when building outside outside of WASM. Useful for" OFF)
+
+if(M32_BINARIES OR COMPILE_WASM)
```

Comment:
Btw, when `COMPILE_WASM` is `ON`, `BUILD_WIDTH` is not being added to compile flags for WASM (although it not even required).

Reply:
So what I did is I replaced all `-m32` and `-m64` with `BUILD_WIDTH`. It will appear wherever those were before. If you notice,  the diff, I have only ever replaced, never placed it in a new place.

Reply:
What I wanted to say is that instead of

`if(M32_BINARIES OR COMPILE_WASM)`

it could have been

`if(M32_BINARIES)`

because WASM is always 32-bit. Just saying :)

Reply:
Yes, true, but I wanted to remove all places where `-m32` or `-m64` was manually defined. 

Also, there are two codepaths for the wasm build, one of which is protected by an `if(0)`. Not sure if someone intends to fix it later: https://github.com/browsermt/marian-dev/blob/8ddb73fad1001ae4c1697d2514ac1e5bd43e2ed3/CMakeLists.txt#L240

Reply:
I intend to use the path that is currently disabled because tinkering with CMAKE_C**_FLAGS is not recommended. Marian does it for native builds but at least for wasm builds, I plan to not follow this practice 👍